apps/frontend/src/components/TodayTextArea.tsx

```diff
@@ -1,99 +1,99 @@
+  const formattedDate = formatDate(selectedDate)
+  const { journal, fetchJournal } = useJournal(formattedDate)
+  const { session } = useAuth()
+  const lastSavedContent = useRef(content)
 
   const handleContentChange = useCallback((newContent: string) => {
-    setContent(newContent);
-    setHasUnsavedChanges(true);
-    setIsSaved(false);
-  }, []);
+    setContent(newContent)
+    setHasUnsavedChanges(true)
+    setIsSaved(false)
+  }, [])
 
   const editor = useEditorHook({
     content,
     setContent: handleContentChange,
     setIsSaved,
-  });
+  })
 
   useEffect(() => {
-    fetchJournal();
-  }, [formattedDate, fetchJournal]);
+    fetchJournal()
+  }, [formattedDate, fetchJournal])
 
   useEffect(() => {
     if (journal?.journal?.content) {
-      setContent(journal.journal.content);
-      editor?.commands.setContent(journal.journal.content);
-      lastSavedContent.current = journal.journal.content;
+      setContent(journal.journal.content)
+      editor?.commands.setContent(journal.journal.content)
+      lastSavedContent.current = journal.journal.content
     } else {
-      setContent("<p></p>");
-      editor?.commands.setContent("<p></p>");
-      lastSavedContent.current = "<p></p>";
+      setContent("<p></p>")
+      editor?.commands.setContent("<p></p>")
+      lastSavedContent.current = "<p></p>"
     }
-    setHasUnsavedChanges(false);
-    setIsSaved(true);
-  }, [journal, editor]);
+    setHasUnsavedChanges(false)
+    setIsSaved(true)
+  }, [journal, editor])
 
   const saveJournal = async () => {
-    if (content === lastSavedContent.current) return;
-    setIsLoading(true);
-    setError(null);
+    if (content === lastSavedContent.current) return
+    setIsLoading(true)
+    setError(null)
     try {
       await axios.post(
         `${BACKEND_URL}/api/journals/create-update/`,
         {
           date: formattedDate,
-          content: content
+          content: content,
         },
         {
           headers: {
-            Authorization: `Bearer ${session}`
-          }
+            Authorization: `Bearer ${session}`,
+          },
         }
-      );
-      setIsSaved(true);
-      setHasUnsavedChanges(false);
-      lastSavedContent.current = content;
+      )
+      setIsSaved(true)
+      setHasUnsavedChanges(false)
+      lastSavedContent.current = content
     } catch (error) {
-      setError("Failed to save journal. Please try again.");
+      setError("Failed to save journal. Please try again.")
     } finally {
-      setIsLoading(false);
+      setIsLoading(false)
     }
-  };
+  }
 
   useEffect(() => {
     if (hasUnsavedChanges && !isLoading) {
-      const debounceTimer = setTimeout(saveJournal, 2000);
-      return () => clearTimeout(debounceTimer);
+      const debounceTimer = setTimeout(saveJournal, 2000)
+      return () => clearTimeout(debounceTimer)
     }
-  }, [content, hasUnsavedChanges, isLoading]);
+  }, [content, hasUnsavedChanges, isLoading])
```

Comment:
Can you reduce the number of useEffects in one component. Too many in a single component can cause isssues

Reply:
oh thanks! actually this page is not mine. this came as file change probably because of my formatter when I did a merge.